ctapipe/image/tests/test_charge_extraction.py

```diff
@@ -12,95 +14,96 @@
 )
 
 
-def test_full_integration(example_event):
-    telid = list(example_event.r0.tel)[0]
-    data = example_event.r0.tel[telid].waveform
-    nsamples = data.shape[2]
-    ped = example_event.mc.tel[telid].pedestal
-    data_ped = data - np.atleast_3d(ped / nsamples)
-    data_ped = np.array([data_ped[0], data_ped[0]])  # Test LG functionality
+@pytest.fixture(scope='module')
+def camera_waveforms():
+    camera = CameraGeometry.from_name("CHEC")
 
-    integrator = FullIntegrator()
-    integration, peakpos, window = integrator.extract_charge(data_ped)
+    n_pixels = camera.n_pixels
+    n_samples = 96
+    mid = n_samples // 2
+    pulse_sigma = 6
+    r_hi = np.random.RandomState(1)
+    r_lo = np.random.RandomState(2)
+
+    x = np.arange(n_samples)
+
+    # Randomize times
+    t_pulse_hi = r_hi.uniform(mid - 10, mid + 10, n_pixels)[:, None]
+    t_pulse_lo = r_lo.uniform(mid + 10, mid + 20, n_pixels)[:, None]
+
+    # Create pulses
+    y_hi = norm.pdf(x, t_pulse_hi, pulse_sigma)
+    y_lo = norm.pdf(x, t_pulse_lo, pulse_sigma)
+
+    # Randomize amplitudes
+    y_hi *= r_hi.uniform(100, 1000, n_pixels)[:, None]
+    y_lo *= r_lo.uniform(100, 1000, n_pixels)[:, None]
+
+    y = np.stack([y_hi, y_lo])
 
+    return y, camera
 
-def test_simple_integration(example_event):
-    telid = list(example_event.r0.tel)[0]
 
-    data = example_event.r0.tel[telid].waveform
-    nsamples = data.shape[2]
-    ped = example_event.mc.tel[telid].pedestal
-    data_ped = data - np.atleast_3d(ped / nsamples)
-    data_ped = np.array([data_ped[0], data_ped[0]])  # Test LG functionality
+def test_full_integration(camera_waveforms):
+    waveforms, _ = camera_waveforms
+    integrator = FullIntegrator()
+    integration, _, _ = integrator.extract_charge(waveforms)
+
+    assert_allclose(integration[0][0], 545.945, rtol=1e-3)
+    assert_allclose(integration[1][0], 970.025, rtol=1e-3)
```

Comment:
can one explain where these numbers come from? Could it help future readers of this ?

Reply:
These values are just the charge extracted for the first pixel in each channel for this ChargeExtractor

Reply:
Right .. erm .. how were they extracted? With the same algorithm we are testing? If so this is somehow circular, no? (Sometimes this is the best we can do, I think)

Reply:
Yeah, it is a bit. They are put here to simply test against future commits, to make sure the expected values do not change, or at least make the contributor aware that their changes has perhaps had an unexpected impact further down the chain.

Reply:
thx